From 45acc0aa9e9c34e635ab56286266e06641600aa6 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:14:22 -0400
Subject: [alsa] hda: Apply +5dB output amp on ASUS Zenbook UX31A

Message-id: <140972823573-42-patchset-jkysela@redhat.com>
Patchwork-id: 90742
O-Subject: [RHEL 7.1 PATCH 041/442] ALSA: hda - Apply +5dB output amp on ASUS Zenbook UX31A
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

ASUS Zenbook UX31A has yet another problem -- softer output level than
others.  According to the measurement, the peak output difference
between 31A and 31E is 5dB.  As ALC269VB has a COEF for the class-D
pre-amp, let's apply it for +5dB.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 2387083157b858f0300f40e629b0fea30f4132c9)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index f9b22fb..c435ae0 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -3833,6 +3833,7 @@ enum {
  ALC269_FIXUP_ACER_AC700,
  ALC269_FIXUP_LIMIT_INT_MIC_BOOST,
  ALC269VB_FIXUP_ASUS_ZENBOOK,
+ ALC269VB_FIXUP_ASUS_ZENBOOK_UX31A,
  ALC269_FIXUP_LIMIT_INT_MIC_BOOST_MUTE_LED,
  ALC269VB_FIXUP_ORDISSIMO_EVE2,
  ALC283_FIXUP_CHROME_BOOK,
@@ -4126,6 +4127,17 @@ static const struct hda_fixup alc269_fixups[] = {
   .chained = true,
   .chain_id = ALC269VB_FIXUP_DMIC,
  },
+ [ALC269VB_FIXUP_ASUS_ZENBOOK_UX31A] = {
+  .type = HDA_FIXUP_VERBS,
+  .v.verbs = (const struct hda_verb[]) {
+   /* class-D output amp +5dB */
+   { 0x20, AC_VERB_SET_COEF_INDEX, 0x12 },
+   { 0x20, AC_VERB_SET_PROC_COEF, 0x2800 },
+   {}
+  },
+  .chained = true,
+  .chain_id = ALC269VB_FIXUP_ASUS_ZENBOOK,
+ },
  [ALC269_FIXUP_LIMIT_INT_MIC_BOOST_MUTE_LED] = {
   .type = HDA_FIXUP_FUNC,
   .v.func = alc269_fixup_limit_int_mic_boost,
@@ -4282,7 +4294,7 @@ static const struct snd_pci_quirk alc269_fixup_tbl[] = {
  SND_PCI_QUIRK(0x1043, 0x106d, "Asus K53BE", ALC269_FIXUP_LIMIT_INT_MIC_BOOST),
  SND_PCI_QUIRK(0x1043, 0x115d, "Asus 1015E", ALC269_FIXUP_LIMIT_INT_MIC_BOOST),
  SND_PCI_QUIRK(0x1043, 0x1427, "Asus Zenbook UX31E", ALC269VB_FIXUP_ASUS_ZENBOOK),
- SND_PCI_QUIRK(0x1043, 0x1517, "Asus Zenbook UX31A", ALC269VB_FIXUP_ASUS_ZENBOOK),
+ SND_PCI_QUIRK(0x1043, 0x1517, "Asus Zenbook UX31A", ALC269VB_FIXUP_ASUS_ZENBOOK_UX31A),
  SND_PCI_QUIRK(0x1043, 0x16e3, "ASUS UX50", ALC269_FIXUP_STEREO_DMIC),
  SND_PCI_QUIRK(0x1043, 0x1a13, "Asus G73Jw", ALC269_FIXUP_ASUS_G73JW),
  SND_PCI_QUIRK(0x1043, 0x1b13, "Asus U41SV", ALC269_FIXUP_INV_DMIC),
-- 
1.7.1